partiql-ast/src/main/java/org/partiql/ast/Insert.java

```diff
@@ -0,0 +1,72 @@
+package org.partiql.ast;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * This is the insert statement.
+ * @see InsertColumnList
+ * @see InsertSource
+ */
+@Builder(builderClassName = "Builder")
+@EqualsAndHashCode(callSuper = false)
+public final class Insert extends Statement {
```

Comment:
There's no need, since there is no state to keep track of. That being said, we can always add that in the future if we find that some state or shared functionality is required.

Reply:
Just noticed that all of the DML statements added in this PR have an `IdentifierChain tableName` so perhaps there exists some shared functionality here? If we were to add a `DML` abstract class that these statements share, then we could add a `tableName` field as a possible modeling.

Reply:
I'm going to defer adding tableName, since I don't see any practical purpose to it at the moment and it can be added without breaking any APIs. Even adding a top-level DML node I'm not entirely convinced of. Especially considering the `RETURNING clause` that I referened earlier on. Yes, it's manipulates data. But it also queries data.

Reply:
Ok, I'm fine w/ deferring. I'll bring up next week w/ the team since we should either
1. make DDL and DML modeling consistent w.r.t. whether there is a top-level classification or
2. make it clear why DDL has a DDL node and why DML does not